src/libNode/Node.cpp

```diff
@@ -1541,11 +1529,116 @@ bool Node::ProcessDoRejoin(const std::vector<unsigned char>& message,
   return true;
 }
 
+void Node::QueryLookupForDSGuardNetworkInfoUpdate() {
+  if (!GUARD_MODE) {
+    LOG_GENERAL(WARNING,
+                "Not in guard mode. Unable to query from lookup for ds guard "
+                "network information update.");
+    return;
+  }
+
+  LOG_MARKER();
+
+  vector<unsigned char> queryLookupForDSGuardNetworkInfoUpdate = {
+      MessageType::LOOKUP,
+      LookupInstructionType::GETGUARDNODENETWORKINFOUPDATE};
+  uint64_t dsEpochNum =
+      m_mediator.m_dsBlockChain.GetLastBlock().GetHeader().GetBlockNum();
+
+  LOG_GENERAL(INFO,
+              "Querying the lookup for any ds guard node network info change "
+              "for ds epoch "
+                  << dsEpochNum);
+
+  if (!Messenger::SetLookupGetNewDSGuardNetworkInfoFromLookup(
+          queryLookupForDSGuardNetworkInfoUpdate, MessageOffset::BODY,
+          m_mediator.m_selfPeer.m_listenPortHost, dsEpochNum)) {
+    LOG_EPOCH(WARNING, to_string(m_mediator.m_currentEpochNum).c_str(),
+              "Messenger::SetLookupGetNewDSGuardNetworkInfoFromLookup failed.");
+    return;
+  }
+  m_requestedForDSGuardNetworkInfoUpdate = true;
+  m_mediator.m_lookup->SendMessageToRandomLookupNode(
+      queryLookupForDSGuardNetworkInfoUpdate);
+}
+
+bool Node::ProcessDSGuardNetworkInfoUpdate(const vector<unsigned char>& message,
+                                           unsigned int offset,
+                                           [[gnu::unused]] const Peer& from) {
+  if (LOOKUP_NODE_MODE) {
+    LOG_GENERAL(
+        WARNING,
+        "Node::ProcessDSGuardNetworkInfoUpdate not expected to be called from "
+        "LookUp node.");
+    return true;
+  }
+
+  if (!GUARD_MODE) {
+    LOG_GENERAL(WARNING,
+                "Not in guard mode. Unable to process from lookup for ds guard "
+                "network information update.");
+    return false;
+  }
+
+  if (!m_requestedForDSGuardNetworkInfoUpdate) {
+    LOG_GENERAL(WARNING,
+                "Did not request for DS Guard node network info update");
+    return false;
+  }
+
+  LOG_MARKER();
+
+  vector<DSGuardUpdateStruct> vecOfDSGuardUpdateStruct;
+  PubKey lookupPubkey;
+  if (!Messenger::SetNodeGetNewDSGuardNetworkInfo(
+          message, offset, vecOfDSGuardUpdateStruct, lookupPubkey)) {
+    LOG_EPOCH(WARNING, to_string(m_mediator.m_currentEpochNum).c_str(),
+              "Messenger::SetNodeGetNewDSGuardNetworkInfo failed.");
+    return false;
+  }
+
+  if (!Lookup::VerifyLookupNode(m_mediator.m_lookup->GetLookupNodes(),
```

Comment:
Since we already verify the lookup signature inside `SetNodeGetNewDSGuardNetworkInfo`, do we still need to do this part?

Reply:
One is to verify the signature is signed by a key that the lookup claimed to be (verify via pubkey). The other is to check whether the pubkey belongs to the any of the known lookup or not.

Reply:
Ok, agree